packages/opentelemetry-node/test/NodeTracerProvider.test.ts

```diff
@@ -160,25 +165,29 @@ describe('NodeTracerProvider', () => {
         logger: new NoopLogger(),
       });
 
-      const sampledParent = provider
-        .getTracer('default')
-        .startSpan('not-sampled-span', {
-          parent: {
-            traceId: 'd4cda95b652f4a1592b449d5929fda1b',
-            spanId: '6e0c63257de34c92',
-            traceFlags: TraceFlags.NONE,
-          },
-        });
+      const sampledParent = provider.getTracer('default').startSpan(
+        'not-sampled-span',
+        {},
+        setExtractedSpanContext(ROOT_CONTEXT, {
+          traceId: 'd4cda95b652f4a1592b449d5929fda1b',
+          spanId: '6e0c63257de34c92',
+          traceFlags: TraceFlags.NONE,
+        })
+      );
       assert.ok(sampledParent instanceof Span);
       assert.strictEqual(
         sampledParent.context().traceFlags,
         TraceFlags.SAMPLED
       );
       assert.strictEqual(sampledParent.isRecording(), true);
 
-      const span = provider.getTracer('default').startSpan('child-span', {
-        parent: sampledParent,
-      });
+      const span = provider
+        .getTracer('default')
+        .startSpan(
+          'child-span',
+          {},
+          setActiveSpan(ROOT_CONTEXT, sampledParent)
```

Comment:
here the same why settings activee span, instead of passing context only 
`sampledParent.context()`

Reply:
Sending parent context directly is now forbidden by the spec.

Reply:
The spec says about not allowing the span to be provided as context
It allows to provide a full context which is `sampledParent.context()` or anything else that will be a full context right ?

```
The API MUST accept the following parameters:
...
1. **The parent Context** or an indication that the new Span should be a root Span. 
2. The API MAY also have an option for implicitly using the current Context as parent as a default behavior. 
3. This API MUST NOT accept a Span or SpanReference as parent, only a full Context.
```
2 - is our  default anyway if you simply omit the context


Setting the sampledParent as active span just to return the context is rather wrong, what if you want to create 5 spans each of them with different context? Will you each time set the desired parent context as active span ??

Reply:
(1) refers to Context, not SpanContext. Setting a SpanContext directly as parent would violate (3).

>what if you want to create 5 spans each of them with different ~~context~~ [parent]?

Notice, the contexts are never set as active. They are just passed directly.

```typescript
const span1 = tracer.startSpan('name1', {}, setActiveSpan(context.active(), parent1));
const span2 = tracer.startSpan('name2', {}, setActiveSpan(context.active(), parent2));
const span3 = tracer.startSpan('name3', {}, setActiveSpan(context.active(), parent3));
const span4 = tracer.startSpan('name4', {}, setActiveSpan(context.active(), parent4));
const span5 = tracer.startSpan('name5', {}, setActiveSpan(context.active(), parent5));
```

Reply:
but there is no point in calling extra `setActiveSpan`

Reply:
The signature is `setActiveSpan(ctx: SpanContext): Context`. `setActiveSpan` takes a `SpanContext` (forbidden for use directly as a parent by (3)), and converts it to a `Context` which is passed as the parent `Context` (as specified by (1)).

Reply:
This is just sugar for `const span1 = tracer.startSpan('name1', {}, context.active().setValue(EXTRACTED_SPAN_CONTEXT_KEY, parent1));`

Reply:
I also find this terrible confusing that one has to set a span either as `active` or `extracted` to get it linked even if it was never extracted nor should it be set as active.
Please note that putting something into the context may result that it gets propagated by context manager. So users which don't want their span set as `active` now have the risk that context manager picks this up and propagates.

Reply:
>I also find this terrible confusing that one has to set a span either as active or extracted to get it linked even if it was never extracted nor should it be set as active.

This was a conversation in the spec for a long time. Maybe @Oberon00 can shed some light here since he wrote the PR which introduced this.

>Please note that putting something into the context may result that it gets propagated by context manager

It is not set as the active context so there is no risk of this.

Reply:
The quoted spec: 

> This API MUST NOT accept a Span or This API MUST NOT accept a Span or SpanReference as parent, only a full Context.as parent, only a full Context.

Note that SpanReference is not SpanContext again, so the spec explicitly forbids accepting SpanContext. There was a rather long discussion here in open-telemetry/opentelemetry-specification#875 and the related issue. One important argument is that the SpanContext is fully defined in the API and so can't be extended with custom properties, etc. However, propagators or samplers or SpanProcessors or custom SDK implementations might need additional, custom properties to be propagated. Propagators could already set these properties in the Context, but passing a SpanContext would discard any of these additional properties.

Thus we now require always a full Context (a SpanContext is not a Context) in the spec.

@Flarna

> I also find this terrible confusing that one has to set a span either as active or extracted to get it linked even if it was never extracted nor should it be set as active.

Maybe what is confusing here is the function name `setActiveSpan`. In Java this function is called `someContext.with(someSpan)`. Also note that the context passed to setActiveSpan is not modified. Rather, a new Context instance is returned with the given span in it.

Reply:
Don't care how the function is named, the context has to hold the span behind key `active` or `extracted` otherwise tracer will not pick it up. In the end it's just one indirection to wrap the relevant information in a context somehow. As it is in the spec the ship has sailed anyway.

Reply:
The argument against this is that context.active() may contain vital information, which you would lose when just passing the SpanContext. By using context.active(), you tell the SDK that logically, the parent is the current context. However, outside of unit tests, most often you will have the active() context of another async task as parent, if you need to set an explicit parent at all. Thus, this is not merely a useless layer of wrapping, it does add information. What is more, from my experience, this ugliness of having to create a new Context with a given SpanContext occurs mostly only in unit tests but not so much in real code, which usually uses an extracted or implicit parent.

For example, in Dynatrace, for Java and Python, this enables use to store some custom information in the Context which we would previously need to squeeze in the tracestate.
Maybe for JS you can get away with monkey-patching the SpanContext, but I think using Context is cleaner.